java/DoublyLinkedList.java

```diff
@@ -0,0 +1,123 @@
+public class DoublyLinkedList {
+
+  public Node head;
+  public Node tail;
+  public int length = 0;
+
+  private class Node {
+    private Node next;
+    private Node prev;
+    private String data;
+
+    // Constructor
+    public Node(Node initNext, Node initPrev, String initData){
+      this.next = initNext;
+      this.prev = initPrev;
+      this.data = initData;
+    }
+
+    // Getters
+    public Node getNext() {
+      return this.next;
+    }
+    public Node getPrev() {
+      return this.prev;
+    }
+    public String getData() {
+      return this.data;
+    }
+
+    // Setters
+    public void setNext(Node setTo) {
+      this.next = setTo;
+    }
+    public void setPrev(Node setTo) {
+      this.prev = setTo;
+    }
+    public void setData(String setTo) {
+      this.data = setTo;
+    }
+  } // End of Node class
+
+  public String insertNode(String data){
+    Node expo = new Node (null, null, data);
```

Comment:
What does `expo` stand for? Curious about the naming here. My rule of thumb is to prefer full words instead of abbreviations, unless the abbreviation is standardized/convention.